src/Config/Converters/McpOptionsConverterFactory.cs

```diff
@@ -0,0 +1,169 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Azure.DataApiBuilder.Config.ObjectModel;
+
+namespace Azure.DataApiBuilder.Config.Converters;
+
+/// <summary>
+/// JSON converter factory for DmlToolsConfig that handles both boolean and object formats.
+/// </summary>
+internal class McpOptionsConverterFactory : JsonConverterFactory
```

Comment:
nit: to be consistent with other factory names, can we rename this to `DmlToolsConfigConverterFactory`?